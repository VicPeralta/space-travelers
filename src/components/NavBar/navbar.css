nav {
  justify-content: space-between;
  padding: 2% 5%;
  border-bottom: 1px solid var(--badge-border-color);
}

.header {
  align-items: center;
  gap: 10px;
}

nav img {
  width: 50px;
}

.navlinks {
  gap: 8px;
  align-items: center;
}

.navlinks a {
  text-decoration: none;
  color: blue;
}

.navlinks a:last-child {
  border-left: 2px solid black;
  padding-left: 8px;
}

.navlinks a.active {
  text-decoration: underline;
}
