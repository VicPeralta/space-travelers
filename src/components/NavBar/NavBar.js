import './navbar.css';
import { NavLink } from 'react-router-dom';
import logo from '../../assets/logo.png';

const links = [
  {
    id: 1,
    path: '/',
    text: 'Rockets',
  },
  {
    id: 2,
    path: '/missions',
    text: 'Missions',
  },
  {
    id: 3,
    path: '/profile',
    text: 'My Profile',
  },
];

const NavBar = () => (
  <nav className="flex">
    <div className="header flex">
      <img src={logo} alt="corporate logo" />
      <h1>Space Travelers&apos; Hub</h1>
    </div>
    <div className="navlinks flex">
      {
      links.map((link) => (
        <NavLink
          key={link.id}
          to={link.path}
        >
          {link.text}
        </NavLink>
      ))
    }
    </div>
  </nav>
);

export default NavBar;
