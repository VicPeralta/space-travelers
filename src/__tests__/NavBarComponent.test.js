import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NavBar from '../components/NavBar/NavBar';

describe('NavBar Test', () => {
  test('Render NavBar', () => {
    const Element = (
      <HashRouter>
        <NavBar />
      </HashRouter>
    );
    const tree = renderer.create(Element).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('NavBar Navigation', () => {
    const Element = (
      <HashRouter>
        <NavBar />
      </HashRouter>
    );
    render(Element);
    const rocketsLink = screen.getByText('Rockets');
    userEvent.click(rocketsLink);
    expect(rocketsLink.classList.contains('active')).toBeTruthy();
    const missionsLink = screen.getByText('Missions');
    userEvent.click(missionsLink);
    expect(missionsLink.classList.contains('active')).toBeTruthy();
    const profileLink = screen.getByText('My Profile');
    userEvent.click(profileLink);
    expect(profileLink.classList.contains('active')).toBeTruthy();
  });
});
