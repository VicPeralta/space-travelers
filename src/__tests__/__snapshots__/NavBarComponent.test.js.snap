// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Test Render NavBar 1`] = `
<nav
  className="flex"
>
  <div
    className="header flex"
  >
    <img
      alt="corporate logo"
      src="logo.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <div
    className="navlinks flex"
  >
    <a
      aria-current="page"
      className="active"
      href="#/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      className=""
      href="#/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      className=""
      href="#/profile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </div>
</nav>
`;
